dbt/scripts/utils/export.py

```diff
@@ -101,149 +173,190 @@ def export_models(
         f"{', '.join(model['name'] for model in models)}"
     )
 
-    conn = pyathena.connect(
+    cursor = pyathena.connect(
         s3_staging_dir=os.getenv(
             "AWS_ATHENA_S3_STAGING_DIR",
-            "s3://ccao-athena-results-us-east-1",
+            "s3://ccao-athena-results-us-east-1/",
         ),
         region_name=os.getenv("AWS_ATHENA_REGION_NAME", "us-east-1"),
-    )
+        cursor_class=pyathena.pandas.cursor.PandasCursor,
+        # Unload query results to speed up execution times
+    ).cursor(unload=True)
 
+    models_for_export: list[ModelForExport] = []
     for model in models:
         # Extract useful model metadata from the columns we queried in
         # the `dbt list` call above
         model_name = model["name"]
+        tags = model["tags"]
         relation_name = model["relation_name"]
         export_name = model["config"]["meta"].get("export_name") or model_name
         template = model["config"]["meta"].get("export_template") or model_name
 
         # Define inputs and outputs for export based on model metadata
         template_path = os.path.join("export", "templates", f"{template}.xlsx")
-        template_exists = os.path.isfile(template_path)
-        output_path = os.path.join("export", "output", f"{export_name}.xlsx")
 
         print(f"Querying data for model {model_name}")
         query = f"SELECT * FROM {relation_name}"
         if where:
             query += f" WHERE {where}"
         print(f"> {query}")
-        model_df = pd.read_sql(query, conn)
+        model_df = cursor.execute(query).as_pandas()
 
-        # Delete the output file if one already exists
-        pathlib.Path(output_path).unlink(missing_ok=True)
+        models_for_export.append(
+            ModelForExport(
+                name=model_name,
+                tags=tags,
+                df=model_df,
+                export_format=(
+                    model["config"]["meta"].get("export_format", {})
+                ),
+                export_filename=f"{export_name}.xlsx",
+                template_path=template_path,
+            )
+        )
 
-        if template_exists:
-            print(f"Using template file at {template_path}")
-            shutil.copyfile(template_path, output_path)
-        else:
-            print("No template file exists; creating a workbook from scratch")
+    return models_for_export
+
+
+def save_model_to_workbook(
+    model: ModelForExport,
+    output_dir: str | None = None,
+) -> pathlib.Path:
```

Comment:
This interface in particular feels a bit brittle to me, since in order to add or change a parameter you have to edit the `ModelForExport` interface as well. A caller also has to understand `ModelForExport` and construct if not calling `query_models_for_export` first or if modifying query results between the `query` and the `save` state. Not perfect, but it was the best incremental design improvement I could think of.